<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chatbot</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2" columnstretch="3,1">
    <item row="1" column="0">
     <layout class="QGridLayout" name="centralGridLayout">
      <item row="0" column="0">
       <widget class="QTabWidget" name="mainTabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="trainTab">
         <attribute name="title">
          <string>Teach</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QSplitter" name="teachTabsplitter">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="handleWidth">
             <number>10</number>
            </property>
            <widget class="QWidget" name="verticalLayoutWidget">
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QHBoxLayout" name="rulesToolbarLayout">
                <item>
                 <widget class="QToolButton" name="addCategoryButton">
                  <property name="font">
                   <font>
                    <pointsize>8</pointsize>
                   </font>
                  </property>
                  <property name="toolTip">
                   <string>Add category</string>
                  </property>
                  <property name="icon">
                   <iconset resource="../resources.qrc">
                    <normaloff>:/icons/add_category_32x32.png</normaloff>:/icons/add_category_32x32.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="addRuleButton">
                  <property name="font">
                   <font>
                    <pointsize>8</pointsize>
                   </font>
                  </property>
                  <property name="toolTip">
                   <string>Add rule</string>
                  </property>
                  <property name="icon">
                   <iconset resource="../resources.qrc">
                    <normaloff>:/icons/add_rule_32x32.png</normaloff>:/icons/add_rule_32x32.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="rmItemButton">
                  <property name="font">
                   <font>
                    <pointsize>8</pointsize>
                   </font>
                  </property>
                  <property name="toolTip">
                   <string>Remove rule or category</string>
                  </property>
                  <property name="icon">
                   <iconset resource="../resources.qrc">
                    <normaloff>:/icons/remove_32x32.png</normaloff>:/icons/remove_32x32.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="rulesToolbarSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::MinimumExpanding</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>0</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QTreeView" name="categoriesTree">
                <property name="mouseTracking">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="verticalLayoutWidget_2">
             <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0,1,0">
              <item>
               <widget class="QLabel" name="categoryNameLabel">
                <property name="text">
                 <string>Category name:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="categoryNameTextEdit"/>
              </item>
              <item>
               <widget class="RuleInputWidget" name="ruleInputWidget" native="true">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>20</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="chatbotRepliesLabel">
                <property name="text">
                 <string>Chatbot replies:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="RuleOutputWidget" name="ruleOutputWidget">
                <property name="mouseTracking">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>0</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="chatTab">
         <attribute name="title">
          <string>Connect with chat</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="1">
           <layout class="QFormLayout" name="chatTabFormLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <property name="leftMargin">
             <number>20</number>
            </property>
            <property name="topMargin">
             <number>20</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="connectWithChatLabel">
              <property name="font">
               <font>
                <weight>50</weight>
                <bold>false</bold>
               </font>
              </property>
              <property name="text">
               <string>Connect with chat:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QRadioButton" name="fbChatRadio">
              <property name="text">
               <string>Facebook</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QRadioButton" name="gtalkChatRadio">
              <property name="text">
               <string>Gmail or Google Talk</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <spacer name="chatTabSpacer2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="usernameLabel">
              <property name="text">
               <string>Username:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="usernameText">
              <property name="maximumSize">
               <size>
                <width>150</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="passwordLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Password:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QLineEdit" name="passwordText">
              <property name="maximumSize">
               <size>
                <width>150</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="echoMode">
               <enum>QLineEdit::Password</enum>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QPushButton" name="connectButton">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Connect</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="conversationsTab">
         <attribute name="title">
          <string>Conversations</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="0">
           <layout class="QGridLayout" name="conversationTabGridLayout" columnstretch="2,3">
            <item row="0" column="0" colspan="2">
             <widget class="QLabel" name="conversationHeaderLabel">
              <property name="font">
               <font>
                <weight>50</weight>
                <bold>false</bold>
               </font>
              </property>
              <property name="text">
               <string>Conversations held with the chatbot:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QTableView" name="conversationContactsTable">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::SingleSelection</enum>
              </property>
              <property name="selectionBehavior">
               <enum>QAbstractItemView::SelectRows</enum>
              </property>
              <property name="sortingEnabled">
               <bool>true</bool>
              </property>
              <attribute name="verticalHeaderCascadingSectionResizes">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderDefaultSectionSize">
               <number>0</number>
              </attribute>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QTableView" name="conversationTable">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="editTriggers">
               <set>QAbstractItemView::NoEditTriggers</set>
              </property>
              <property name="alternatingRowColors">
               <bool>true</bool>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::NoSelection</enum>
              </property>
              <property name="sortingEnabled">
               <bool>true</bool>
              </property>
              <attribute name="verticalHeaderCascadingSectionResizes">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderDefaultSectionSize">
               <number>0</number>
              </attribute>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <widget class="QTabWidget" name="rightSideTabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="testTab">
       <attribute name="title">
        <string>Test your chatbot</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="testToolbarLayout">
          <item>
           <widget class="QToolButton" name="clearTestConversationButton">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="testToolbarSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="testConversationText">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="overwriteMode">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="testLabel">
          <property name="text">
           <string>Test:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="testInputText"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RuleOutputWidget</class>
   <extends>QPlainTextEdit</extends>
   <header>ruleoutputwidget.h</header>
  </customwidget>
  <customwidget>
   <class>RuleInputWidget</class>
   <extends>QWidget</extends>
   <header>ruleinputwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
</ui>
